aleph/oauth.py

```diff
@@ -77,6 +82,65 @@ def handle_google_oauth(provider, oauth_token):
     return Role.load_or_create(user_id, Role.USER, data.get('name'),
                                email=data.get('email'))
 
+def get_claims_cognito(token,keys,verify_aud=False):
+    headers = jwt_jose.get_unverified_headers(token)
+    kid = headers['kid']
+    key_index = -1
+    for i in range(len(keys)):
+        if kid == keys[i]['kid']:
+            key_index = i
+            break
+    if key_index == -1:
+        log.warning('Public key not found in jwks.json')
+        return False
+    #Construct Public Key
+    public_key = jwk.construct(keys[key_index])
+    message, encoded_signature = str(token).rsplit('.', 1)
+    decoded_signature = base64url_decode(encoded_signature.encode('utf-8'))
+
+    if not public_key.verify(message.encode("utf8"), decoded_signature):
+        log.warning('Signature verification failed')
+        return False
+
+    claims = jwt_jose.get_unverified_claims(token)
```

Comment:
Can this use: https://docs.authlib.org/en/latest/jose/jwt.html#jwt-payload-claims-validation ?